addOns/accessControl/src/main/java/org/zaproxy/zap/extension/accessControl/AccessControlAPI.java

```diff
@@ -0,0 +1,217 @@
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_SCAN,
+                        new String[] {PARAM_CONTEXT_ID, PARAM_USER_ID},
+                        new String[] {
+                            PARAM_UNAUTH_USER, PARAM_RAISE_ALERT, PARAM_ALERT_RISK_LEVEL
+                        }));
+
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_WRITE_HTML_REPORT, new String[] {PARAM_CONTEXT_ID, PARAM_FILENAME}));
+
+        this.addApiView(new ApiView(VIEW_GET_SCAN_PROGRESS, new String[] {PARAM_CONTEXT_ID}));
+        this.addApiView(new ApiView(VIEW_GET_SCAN_STATUS, new String[] {PARAM_CONTEXT_ID}));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        ApiResponse result = null;
+
+        switch (name) {
+            case ACTION_SCAN:
+                LOGGER.debug("Access control start scan called");
+
+                AccessControlScanStartOptions startOptions = new AccessControlScanStartOptions();
+
+                startOptions.targetContext = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID);
+
+                if (usersExtension == null) {
+                    usersExtension =
+                            Control.getSingleton()
+                                    .getExtensionLoader()
+                                    .getExtension(ExtensionUserManagement.class);
+                }
+
+                List<User> users = new ArrayList<User>();
+
+                String[] commaSeparatedUserIDs =
+                        ApiUtils.getNonEmptyStringParam(params, PARAM_USER_ID).split("\\s*,\\s*");
+
+                for (int i = 0; i < commaSeparatedUserIDs.length; i++) {
+                    int userID = Integer.parseInt(commaSeparatedUserIDs[i]);
+                    users.add(
+                            usersExtension
+                                    .getContextUserAuthManager(
+                                            startOptions.targetContext.getIndex())
+                                    .getUserById(userID));
+                }
+
+                startOptions.targetUsers = users;
+
+                // Add unauthenticated user
+                if (params.has(PARAM_UNAUTH_USER)
+                        && StringUtils.isNotBlank(params.getString(PARAM_UNAUTH_USER))
+                        && params.getBoolean(PARAM_UNAUTH_USER)) {
+                    startOptions.targetUsers.add(null);
+                }
+
+                if (params.has(PARAM_RAISE_ALERT)
+                        && StringUtils.isNotBlank(params.getString(PARAM_RAISE_ALERT))) {
+                    startOptions.raiseAlerts = params.getBoolean(PARAM_RAISE_ALERT);
+                } else {
+                    startOptions.raiseAlerts = true;
+                }
+
+                if (params.has(PARAM_ALERT_RISK_LEVEL)
```

Comment:
Can be `startOptions.alertRiskLevel = params.optInt(PARAM_ALERT_RISK_LEVEL, Alert.RISK_HIGH);`